node/tests/reconstruct_test.go

```diff
@@ -0,0 +1,120 @@
+//go:build !race
```

Comment:
😸 
reminds me of [this PR's try](https://github.com/celestiaorg/celestia-node/pull/578/files#diff-4c9592dec9e12f125d6ae5f8b1a0be082e3e9b650f55141d9c42038af78cf241R1)

A short note why this is needed would be highly appreciated